### PR TITLE
section 4 complete to be checked

### PR DESCRIPTION
**Pull request by Jamtartarenice**

### CODE REVIEW

File: src/components/getArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios'
+
+function getArticles(setItem) {
```
Comment (MRombi) [inline src/components/getArticles.jsx]: This can be in a js file instead of jsx 🙂

File: src/components/getArticles.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+        axios.get('https://jakub-news-web.onrender.com/api/articles')
+        .then((response) => {
+            setItem(response.data.articles)
+            console.log(response.data.articles)
```
Comment (MRombi) [inline src/components/getArticles.jsx]: We should remove console.log

File: src/components/ListOfArticles.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import getArticles from "./getArticles";
+import { useState ,useEffect} from "react";
+
+const ListOfArticles = () => {
+    const [articles,setArticles] = useState([]);
+
+    useEffect(() => {    
+        getArticles(setArticles)
+        console.log('it works')
```
Comment (MRombi) [inline src/components/ListOfArticles.jsx]: We should remove console.log

File: src/components/ListOfArticles.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import getArticles from "./getArticles";
+import { useState ,useEffect} from "react";
+
+const ListOfArticles = () => {
+    const [articles,setArticles] = useState([]);
```
Comment (saimakhanom) [inline src/components/ListOfArticles.jsx:7]: we should have a loading state to provide visual feedback to the user on slow load

File: src/App.jsx [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
+
+  useEffect(() => {
+    console.log(name)
+  },[name]);
 
   return (
     <>
-      beans
+      <SearchBar input={input} setInput={setInput} setName={setName}/>
+      <ListOfArticles />
```
Comment (MRombi) [inline src/App.jsx]: We should have routing 🙂

File: src/App.jsx [hunk 6]
```diff
@@ -1,12 +1,21 @@
...
 
 function App() {
-  const [count, setCount] = useState(0)
+
+  const [input,setInput] = useState('');
+  const [name, setName] = useState('');
+
+  useEffect(() => {
+    console.log(name)
+  },[name]);
```
Comment (MRombi) [inline src/App.jsx]: We should have a loading state, ideally a loading component for reusability 🙂

File: src/components/ListOfArticles.jsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    useEffect(() => {    
+        getArticles(setArticles)
+        console.log('it works')
+    },[]);
+
+    return <ul>
+        {articles.map(article => <li className='InListOfArticles' key={article.article_id}>{article.title}</li> )}
```
Comment (saimakhanom) [inline src/components/ListOfArticles.jsx]: we want to see a bit more information for our articles in the articles list, so votes, date create (in a human readable format), number of votes and comment count. see this article for the best way to display these: https://www.refactoringui.com/previews/labels-are-a-last-resort

File: src/components/getArticles.jsx (hunk 1 above)
Comment (saimakhanom) [inline src/components/getArticles.jsx]: this should go into a utils folder as a  .js file, but well done for extracting it out to its own file!

File: src/components/ListOfArticles.jsx (hunk 7 above)
Comment (MRombi) [inline src/components/ListOfArticles.jsx]: We should display more than the title 🙂

**Review by MRombi — CHANGES_REQUESTED** [review]
Good job on your first attempt with this ticket! Please action on the feedback I left and resubmit 🙂

File: src/App.jsx [hunk 8]
```diff
@@ -1,22 +1,14 @@
...
-    console.log(name)
-  },[name]);
-
   return (
-    <>
-      <SearchBar input={input} setInput={setInput} setName={setName}/>
-      <ListOfArticles />
-    </>
+    <Routes>
+      <Route path='' element={<Home />} />
```
Comment (saimakhanom) [inline src/App.jsx:10]: the path should be '/'

File: src/components/ListOfArticles.jsx [hunk 9]
```diff
@@ -1,17 +1,24 @@
...
 
-    return <ul>
-        {articles.map(article => <li className='InListOfArticles' key={article.article_id}>{article.title}</li> )}
-    </ul>
+    return <Loading>
+        <ul>
+            {articles.map(article => <li className='InListOfArticles' key={article.article_id}>
+                <div className="Title">{article.title} </div> 
+                <div className="DateOfPost">{formatDate(article.created_at)}</div>
+                <div className="VotesOfPost">{article.votes} votes</div>
```
Comment (saimakhanom) [inline src/components/ListOfArticles.jsx:18]: i would also like to see comment_count here as well as author

**Pull request merged**